lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -500,10 +500,23 @@ def authorize_ip(type, changed, client, group, groupRules,
                             client.authorize_security_group_egress(GroupId=group['GroupId'],
                                                                    IpPermissions=[ip_permission])
                     except botocore.exceptions.ClientError as e:
-                        module.fail_json(msg="Unable to authorize %s for ip %s security group '%s' - %s" %
-                                             (type, thisip, group['GroupName'], e),
-                                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
-            changed = True
+                        if e.response['Error']['Code'] == "InvalidPermission.Duplicate":
+                            # There are host bits but only the network bits get authorized so it appears there is a conflict.
+                            # Allow graceful completion with a warning.
+                            actual_ip = re.search(r"(?<!\d\.)(?<!\d)(?:\d{1,3}\.){3}\d{1,3}/\d{1,2}(?!\d|(?:\.\d))", e.response['Error']['Message']).group()
```

Comment:
This regex is going to break for cases where the rule is against an IPv6 CIDR range